starlette/responses.py

```diff
@@ -289,7 +289,7 @@ def set_stat_headers(self, stat_result: os.stat_result) -> None:
         etag_base = str(stat_result.st_mtime) + "-" + str(stat_result.st_size)
         etag = hashlib.md5(etag_base.encode()).hexdigest()
 
-        self.headers.setdefault("content-length", content_length)
+        self.headers["content-length"] = content_length
```

Comment:
We are sure that `content-length` exists at this point, considering the change of conditional in the `init_headers()`.

Reply:
I'd be a bit careful about that assumption. Personally I'd *start* by reverting this bit, making sure we've got the different test cases I've described. The `StreamingResponse` and `FileResponse` won't yet pass, but let's look at *how* we should resolve them once we've identified the test failures fully.